### DISCUSSION TITLE
DALY UART connection

### DISCUSSION THREAD

**Discussion by smejfi** [Louisvdw/dbus-serialbattery#89] [Q&A]
Hello, I have a question about connecting DALY smart bms with uart. Im using original VE direct to USB cable and I cant get Venus os to see the bms although when I connect it to PC (with the same cable) and use the BMS Monitor sw from Daly, it works perfectly. Also, when I use Daly RS485 to USB cable, it works well and I can see it in Venus OS. Would you please have some advise what could be the problem? The reason why I want to use UART is because eventualy I need to use different bms which uses xiaoxiang app and has only UART port. I tried that one first and it didnt work so I figured I would try it with daly bms (which I know is working at least with RS485) so I would find out if the problem is in the bms or somewhere else and for now it seems that the problem is not in bms because daly with uart doesnt work either. Thank you for your answer.

**Comment by Louisvdw** [comment]
Hi @smejfi 
There are 3 possible problems that you can check.

1. Make sure the cable you use is compatible with the BMS. The common UART types you get are TTL (sometimes just labled UART), RS232(also sometime just labled UART) and RS485. They each work with different voltages and different protocols. Ve.Direct is TTL cable.
2. The Ve.Direct cable is isolated meaning power cannot flow from the one side to the other. It's a safety feature but it require that you supply power from both sides. The USB side gets power from the USB and you also need to connect the Ground and Positive from the BMS side (normally you will supply 5V).
3. The RX/TX cables can be the wrong way around. It is say to just swop them to see if it will work.

If your Daly has a RS485 cable just use that one.
The UART port on the JBD/LLT BMS is a TTL and the Ve.Direct should work if powered from the BMS side as well.
Also look at the Troubleshoot section in the wiki on which log files you need to look at they will tell you what is happening.

**Reply by smejfi** [reply]
Thank you for your answer, Im trying to open the log file using command with putty "/data/log/serial-starter/current" and it says Permission denied, any hints how to overcome this?

**Reply by Louisvdw** [reply]
use a text viewer like nano
`nano /data/log/serial-starter/current`

Or if you just want to see the last few lines
`tail /data/log/serial-starter/current`

**Reply by smejfi** [reply]
Thank you, so, in /data/log/serial-starter/current it shows only "Start service vedirect-interface.ttyUSB1", the line with "daemontools...." is missing. Then in /data/log/dbus-serialbattery.ttyUSB1/current it tries all possible bms types and eventually says "ERROR >>> No battery connection at /dev/ttyUSB1". Any clue why it skips Daly BMS with "no reply"? If I plug this setup to windows pc (instead of venus) and run the daly software, it normally connects and loads the data from bms...

The missing line "Create daemontools service dbus-serialbattery.ttyUSB1" is probably not correct right?
I have also tried different RS232 to USB converter that I found at home and when I connect that one and observ log file "/data/log/serial-starter/current", it is still trying to run it with different devices in the loop again and again, which probably means it is some no brand converter and Venus has no driver for it, right?

**Reply by smejfi** [reply]
One of the possible reasons why the original VE direct cable might not work is that positive voltage on the Daly BMS serial output is only 3.3 V, but then it wouldnt make sense that the comunication works on windows pc with the same cable...

**Reply by Louisvdw** [reply]
Daly has a few different chips and interfaces that it use for BMS. It might be that your model is using something different to the 2 types the driver currently implement. Please also try the Daly with the RS485 connection.

The JBD BMS all use the same protocol, so your BMS will be supported. You just need your cable connected correctly.

**Reply by smejfi** [reply]
Well, my original plan was to try QUCC BMS (which by the way looks exactly the same as JBD, but I guess there are some differences), I know it is not specifically written in the supported devices, but I thought I would give it a try as it uses Xiaoxiang bt app as well. Do you think it should work?

**Reply by smejfi** [reply]
So, thank you for all your answers! Although it is still unclear to me why the ve direct to usb cable is not working for neither daly, neither qucc, I managed to get them both working with non-isolated uart to usb converter. With daly, the problem was probably somehow related with bms not being activated, because I tried to connect it to the windows computer with this non-isolated uartToUsb, it wasnt working, then I connected it to windows with original Daly 485toUSB cable, it started working, then I connected it to windows again with this non-isolated uartToUsb, it started working as well, and finally I connected it to GX with the same cable and it started working even there. Then I changed Daly to QUCC (still with uart) and it is working as well. Nevertheless, with original VE direct to USB cable I didnt manage to get it working, no idea why.

**Reply by Louisvdw** [reply]
Is your BMS connected to cells, or were you running your BMS without any cells?

**Reply by Louisvdw** [reply]
QUCC does not make BMS. They just sell other BMS. So that one that looks like the JBD is a JBD with another label. I have a QUCC labled bms on the bench that I still need to get to, and that is actually an AXEBMS. This is why they work even though QUCC is not on the supported list.

**Reply by smejfi** [reply]
It is connected to the cells, namely the Daly BMS is connected to 13s li-ion and qucc bms is connected to 4S lifepo4. Actually, I found out that qucc bms uart port has quite large voltage on its VCC pin, I think I measured over 12 V! The first thing I did when the VE direct to USB cable arrived from the shop, I connected it to that qucc UART port (including its VCC pin), so one of my ideas is that I might have damaged that VE direct to USB cable. On the other hand, then I tested it with my MPPT and it was communicating with no issues, so it should be fine...

**Reply by Louisvdw** [reply]
Then it could be that it is not compatible with the BMS.
It could also just be a loose connection messing up the signals if the fit is not great.

**Reply by smejfi** [reply]
Maybe, but with Daly BMS I was able to make it work on windows, so the connection and the cable-bms compatibity should be fine...really, I dont know :-D

**Reply by rbeudel** [reply]
I got my 16S Daly working via UART with a simple FDTI USB converter connected only TX,RX and gnd. I bought a DALY UART cable in a german shop, no function with windows or venus os.
